pkg/storage/chunk/client/ibmcloud/cos_object_client.go

```diff
@@ -135,6 +152,29 @@ func validate(cfg COSConfig) error {
 	if cfg.Endpoint == "" {
 		return errEmptyEndpoint
 	}
+
+	if cfg.ApiKey.String() == "" {
```

Comment:
If we don't have complete HMAC credentials or API Key we should raise an error that there must be at least one authentication method enabled.